com.fasterxml.jackson.databind.JsonMappingException

### Describe the issue

I did not change my flow but now When I try to execute it I got error below: 

Execution failed! With last error: com.fasterxml.jackson.databind.JsonMappingException: Unexpected character ('p' (code 112)): was expecting comma to separate Array entries at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1502] (through reference chain: java.util.ArrayList[1])
2025-02-07 16:32:59.094runetlscript
com.fasterxml.jackson.databind.JsonMappingException: Unexpected character ('p' (code 112)): was expecting comma to separate Array entries
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1502] (through reference chain: java.util.ArrayList[1])



The error happens on runetlscript 

```yaml
- id: fileSystem
    type: io.kestra.plugin.core.flow.WorkingDirectory
    tasks:  
    - id: git
      type: io.kestra.plugin.git.Clone
      url: https://GIT_URL
      branch: "{{ outputs.getBranchNameByNameSpace.vars.value }}"
      username: azure
      password: "{{ secret('GITAZURE_ETL') }}"
          
    - id: runetlscript
      type: io.kestra.plugin.scripts.shell.Commands
      logLevel: INFO
      namespaceFiles: 
        enabled: true
      description: "Run the ETL Python script to transfer data from MSSQL to PostgreSQL."
      taskRunner:
        type: io.kestra.plugin.core.runner.Process         
      commands: 
        - |
          #!/bin/bash
          set -e

          # Function to check if a pip package is installed
          is_package_installed() {
            pip show "$1" > /dev/null 2>&1
          }

          install_pip_package() {
            if ! is_package_installed "$1"; then
                pip install -qq "$1"
            fi
          }

          is_installed() {
            dpkg-query -W -f='${Status}' "$1" 2>/dev/null | grep -q "install ok installed"
          }

          if ! is_installed jq; then
              apt-get update && apt-get install -y jq
          fi

          if ! is_installed msodbcsql18; then
              curl -s https://packages.microsoft.com/keys/microsoft.asc  | tee /etc/apt/trusted.gpg.d/microsoft.asc
              curl s https://packages.microsoft.com/config/ubuntu/22.04/prod.list | tee /etc/apt/sources.list.d/mssql-release.list
              apt-get update && ACCEPT_EULA=Y apt-get install -y msodbcsql18
          fi

          if ! is_installed unixodbc-dev; then
              apt-get update && apt-get install -y unixodbc-dev
          fi
          
          MSSQL_DB=$(echo '{{ inputs.databaseData }}' | jq -r '.mssql_db')
          mssql_dbip=$(echo '{{ inputs.databaseData}}' | jq -r '.mssql_dbip')
          PG_DB=$(echo '{{ inputs.databaseData }}' | jq -r '.pg_dbestado')
          pg_dbip=$(echo '{{ inputs.databaseData }}' | jq -r '.pg_dbip')
          pg_dbport=$(echo '{{ inputs.databaseData}}' | jq -r '.pg_dbport')

          install_pip_package pandas
          install_pip_package sqlalchemy
          install_pip_package psycopg2-binary
          install_pip_package argparse
          install_pip_package pyodbc

          python scripts/mov_cfop/mov_por_cfop.py --mssql_db "$MSSQL_DB" --mssql_dbip "$mssql_dbip" --mssql_dbpass "{{ secret('MSSQLPROD') }}" --pg_db "$PG_DB" --pg_dbip "$pg_dbip" --pg_dbport "$pg_dbport" "--pg_dbpass={{ secret('POSTGREPROD') }}" --id_importacao {{ inputs.id_sig_importacao }} --batch_size 100000
      env:
        ACCEPT_EULA: "Y"
```

### Environment

- Kestra Version: develop

Hi @loicmathieu, @HeribertoAlves ,

I ran into a similar issue with `io.kestra.plugin.scripts.python.Commands` after upgrading to `v0.21.1`:

```
2025-02-07 07:00:13.148 com.fasterxml.jackson.databind.JsonMappingException: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 315] (through reference chain: java.util.ArrayList[0])
```

Stack trace: 

```
2025-02-07T06:00:13.148Z ERROR com.fasterxml.jackson.databind.JsonMappingException: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 315] (through reference chain: java.util.ArrayList[0])
2025-02-07T06:00:13.148Z TRACE io.kestra.core.exceptions.IllegalVariableEvaluationException: com.fasterxml.jackson.databind.JsonMappingException: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 315] (through reference chain: java.util.ArrayList[0])
	at io.kestra.core.models.property.Property.asList(Property.java:146)
	at io.kestra.core.models.property.Property.asList(Property.java:129)
	at io.kestra.core.runners.RunContextProperty.lambda$asList$2(RunContextProperty.java:105)
	at io.kestra.core.utils.Rethrow.lambda$throwFunction$4(Rethrow.java:89)
	at java.base/java.util.Optional.map(Unknown Source)
	at io.kestra.core.runners.RunContextProperty.asList(RunContextProperty.java:105)
	at io.kestra.plugin.scripts.python.Commands.run(Commands.java:261)
	at io.kestra.plugin.scripts.python.Commands.run(Commands.java:22)
	at io.kestra.core.runners.WorkerTaskCallable.doCall(WorkerTaskCallable.java:78)
	at io.kestra.core.runners.AbstractWorkerCallable.call(AbstractWorkerCallable.java:62)
	at io.kestra.core.runners.WorkerSecurityService.callInSecurityContext(WorkerSecurityService.java:10)
	at io.kestra.core.runners.Worker.lambda$callJob$19(Worker.java:837)
	at io.kestra.core.trace.NoopTracer.inCurrentContext(NoopTracer.java:15)
	at io.kestra.core.runners.Worker.callJob(Worker.java:833)
	at io.kestra.core.runners.Worker.runAttempt(Worker.java:792)
	at io.kestra.core.runners.Worker.run(Worker.java:640)
	at io.kestra.core.runners.Worker.handleTask(Worker.java:332)
	at io.kestra.core.runners.Worker.lambda$run$8(Worker.java:261)
	at io.micrometer.core.instrument.internal.TimedRunnable.run(TimedRunnable.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 315] (through reference chain: java.util.ArrayList[0])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:401)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:372)
	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:223)
	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:184)
	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:27)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4917)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3860)
	at io.kestra.core.models.property.Property.asList(Property.java:144)
	... 21 more
Caused by: com.fasterxml.jackson.core.JsonParseException: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 315]
	at com.fasterxml.jackson.core.JsonParser._constructReadException(JsonParser.java:2660)
	at com.fasterxml.jackson.core.base.ParserBase._throwUnquotedSpace(ParserBase.java:1409)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._finishString2(ReaderBasedJsonParser.java:2202)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._finishString(ReaderBasedJsonParser.java:2173)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.getText(ReaderBasedJsonParser.java:295)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextTextValue(ReaderBasedJsonParser.java:1216)
	at com.fasterxml.jackson.databind.deser.std.StringCollectionDeserializer.deserialize(StringCollectionDeserializer.java:203)
	... 27 more

```

Downgrading (back) to `v0.20.14` resolved the issue.

A diff on the docker history of `v0.20.14` and `v.0.20.15` indicates the jdk version changed, could this have had an effect? 

`ENV JAVA_VERSION=jdk-21.0.5+11` (`v0.20.14`) vs `ENV JAVA_VERSION=jdk-21.0.6+7` (`v0.20.15+`)

--- a/core/src/main/java/io/kestra/core/models/property/Property.java
+++ b/core/src/main/java/io/kestra/core/models/property/Property.java
@@ -22,6 +22,8 @@ import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
+import static io.kestra.core.utils.Rethrow.throwFunction;
+
 /**
  * Define a plugin properties that will be rendered and converted to a target type at use time.
  *
@@ -136,12 +138,29 @@ public class Property<T> {
      *
      * @see io.kestra.core.runners.RunContextProperty#asList(Class, Map)
      */
+    @SuppressWarnings("unchecked")
     public static <T, I> T asList(Property<T> property, RunContext runContext, Class<I> itemClazz, Map<String, Object> variables) throws IllegalVariableEvaluationException {
         if (property.value == null) {
-            String rendered =  runContext.render(property.expression, variables);
             JavaType type = MAPPER.getTypeFactory().constructCollectionLikeType(List.class, itemClazz);
             try {
-                property.value = MAPPER.readValue(rendered, type);
+                String trimmedExpression = property.expression.trim();
+                // We need to detect if the expression is already a list or if it's a pebble expression (for eg. referencing a variable containing a list).
+                // Doing that allows us to, if it's an expression, first render then read it as a list.
+                if (trimmedExpression.startsWith("{{") && trimmedExpression.endsWith("}}")) {
+                    property.value = MAPPER.readValue(runContext.render(property.expression, variables), type);
+                }
+                // Otherwise if it's already a list, we read it as a list first then render it from run context which handle list rendering by rendering each item of the list
+                else {
+                    List<?> asRawList = MAPPER.readValue(runContext.render(property.expression, variables), List.class);
+                    property.value = (T) asRawList.stream()
+                        .map(throwFunction(item -> {
+                            if (item instanceof String str) {
+                                return MAPPER.convertValue(str, itemClazz);
+                            }
+                            return item;
+                        }))
+                        .toList();
+                }
             } catch (JsonProcessingException e) {
                 throw new IllegalVariableEvaluationException(e);
             }

--- a/core/src/test/java/io/kestra/core/models/property/PropertyTest.java
+++ b/core/src/test/java/io/kestra/core/models/property/PropertyTest.java
@@ -298,6 +298,30 @@ class PropertyTest {
         assertThat(output.getMap().get("mapKey2"), is("mapValue2"));
     }
 
+    @Test
+    void aListToRender() throws Exception {
+        var task = DynamicPropertyExampleTask.builder()
+            .items(new Property<>("""
+                ["python test.py --input1 \\"{{ item1 }}\\" --input2 \\"{{ item2 }}\\""]"""))
+            .properties(new Property<>("""
+                {
+                  "key1": "{{value1}}",
+                  "key2": "{{value2}}"
+                }"""))
+            .build();
+        var runContext = runContextFactory.of(Map.ofEntries(
+            entry("item1", "item1"),
+            entry("item2", "item2"),
+            entry("value1", "value1"),
+            entry("value2", "value2")
+        ));
+
+        var output = task.run(runContext);
+
+        assertThat(output, notNullValue());
+        assertThat(output.getList(), containsInAnyOrder("python test.py --input1 \"item1\" --input2 \"item2\""));
+    }
+
     @Builder
     @Getter
     private static class TestObj {

